test/libsolidity/semanticTests/enums/enum_referencing.sol

```diff
@@ -0,0 +1,41 @@
+interface I {
+    enum Direction { A, B, Left, Right }
+}
+library L {
+    enum Direction { Left, Right }
+    function f() public pure returns (Direction) {
+        return Direction.Right;
+    }
+    function g() public pure returns (I.Direction) {
+        return I.Direction.Right;
+    }
+}
+contract C is I {
+    function f() public pure returns (Direction) {
+        return Direction.Right;
+    }
+    function g() public pure returns (I.Direction) {
+        return I.Direction.Right;
+    }
+    function h() public pure returns (L.Direction) {
+        return L.Direction.Right;
+    }
+    function x() public pure returns (L.Direction) {
+        return L.f();
+    }
+    function y() public pure returns (I.Direction) {
+        return L.g();
+    }
+}
+// ====
+// compileViaYul: also
+// compileToEwasm: false
+// ----
+// library: L
+// f() -> 3
```

Comment:
Wasn't this 1 in the original test?

Reply:
Ah ok now I see: The original test first calls a function on L itself. I'm not sure if we can do this inside the semantic tests.

Reply:
Removed from this commit and moved to another branch for further work.

Reply:
The commit has been extracted from SolidityEndToEnd.cpp 

Should we insert it back there as cpp code? Or should we merely shift the .sol file to a different folder